pkg/services/object/acl/eacl/v2/headers.go

```diff
@@ -179,44 +186,51 @@ func (h *headerSource) objectHeaders() ([]eaclSDK.Header, bool) {
 
 			oV2.SetHeader(hdr)
 
-			return headersFromObject(object.NewFromV2(oV2), h.addr), true
+			return headersFromObject(object.NewFromV2(oV2), h.addr), nil
 		}
 	}
 
-	return nil, true
+	return nil, nil
 }
 
-func (h *headerSource) localObjectHeaders(addr *objectSDKAddress.Address) ([]eaclSDK.Header, bool) {
+func (h *cfg) localObjectHeaders(addr *objectSDKAddress.Address) ([]eaclSDK.Header, error) {
 	obj, err := h.storage.Head(addr)
-	if err == nil {
-		return headersFromObject(obj, addr), true
+	if err != nil {
+		// Still parse addressHeaders, because the errors is ignored in some places.
+		return addressHeaders(addr), err
 	}
-
-	return addressHeaders(addr), false
+	return headersFromObject(obj, addr), nil
 }
 
-func cidHeader(idCnr *cid.ID) eaclSDK.Header {
-	return &sysObjHdr{
+func cidHeader(idCnr cid.ID) sysObjHdr {
+	return sysObjHdr{
 		k: acl.FilterObjectContainerID,
-		v: cidValue(idCnr),
+		v: idCnr.String(),
```

Comment:
There was an error in `cidValue` (and other ID funcs): we need to use `EncodeToString` since `String` is unstable. Lets fix it in this PR?

Reply:
Fixed.